runtime-modules/project-token/src/types.rs

```diff
@@ -602,43 +616,14 @@ pub(crate) enum AmmOperation {
     Sell,
     Buy,
 }
-impl<Balance: Zero + Copy + Saturating> AmmCurve<Balance> {
-    pub(crate) fn from_params(params: AmmParams) -> Self {
+impl<Balance: TokenBalanceTrait> AmmCurve<Balance> {
+    pub(crate) fn from_params<T: Config>(params: AmmParams<Balance>) -> Self {
         Self {
             slope: params.slope,
             intercept: params.intercept,
             provided_supply: Balance::zero(),
         }
     }
-    pub(crate) fn eval<T: Config>(
```

Comment:
This seemed like a better place for the 'eval' implementation., but it was made into a standalone function in the content pallet. What is the reason behind this?